### PR TITLE
First pass at `split_unknown` function

### PR DESCRIPTION
**Pull request by krpaulson**
## Describe changes

This is a first pass at a "minimum viable product" type version of a `split_unknown` function. I think a lot of work could be done to increase flexibility of this function, in line with some of the functionality of `agg` and `scale` functions in this package.

## What issues are related

Related to #22 

## Checklist

* [x] Have you read the [contributing guidelines](https://github.com/ihmeuw-demographics/packageTemplate/wiki#guide-to-r-package-development) for `ihmeuw-demographics` R packages?
* [ ] Have you successfully run `devtools::check()` locally?
* [x] Have you updated or added function (and vignette if applicable) documentation? Did you update the 'man' and 'NAMESPACE' files with `devtools::document()`?
* [x] Have you added in tests for the changes included in the PR?
* [x] Do the changes follow the `ihmeuw-demographics` [code style](https://github.com/ihmeuw-demographics/packageTemplate/wiki/Code-style-guide)?
* [ ] Do the changes need to be immediately included in a new build of [`docker-base`](https://github.com/ihmeuw-demographics/docker-base) or [`docker-internal`](https://github.com/ihmeuw-demographics/docker-internal)? If so follow directions in those repositories to rebuild and redeploy the images.
* [ ] Do the changes require updates to other repositories which use this package? If yes, make the necessary updates in those repos, and consider integration tests for those repositories.

## Details of PR

I'd like to have a working version of this function good to go soon-ish, but I'm also wondering @chacalle if you're up for spending some time on this to make it more in line with other functions in this package.

### CODE REVIEW

File: R/split.R [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+
+  # distribute --------------------------------------------------------------
+
+  # TODO: do aggregation using mapping input?
+
+  # calculate weights
+  # TODO: perhaps add option to pass in custom weights
+  dt[,
+    weight := get(value_cols) / sum(get(value_cols)[!(unknown)]),
```
Comment (chacalle) [inline R/split.R:112]: Will this work with multiple `value_cols`? Can revisit later.

**Review by chacalle — APPROVED** [review]
This looks good for a first pass. I think the main thing we'll want to revisit is whether this function can be made more flexible to split aggregate groupings (or if we want a separate function for that).

**Comment by chacalle** [comment]
Just need small fixes for the package checks

File: R/split.R (hunk 1 above)
Comment (krpaulson) [reply R/split.R:116]: No not yet

**Pull request merged**